test/integration/MErc20DelegateFixer.t.sol

```
@@ -0,0 +1,805 @@
+    }
+
+    function _enterMarkets(address sender) private {
+        address[] memory mTokens = new address[](1);
+        mTokens[0] = address(delegator);
+
+        vm.startPrank(sender);
+        comptroller.enterMarkets(mTokens);
+        vm.stopPrank();
+    }
+
+    function _liquidityShortfall(address account) private {
+        (uint256 err,, uint256 shortfall) = comptroller.getAccountLiquidity(account);
+
+        assertEq(err, 0);
+        assertEq(shortfall, 0);
+    }
+
+    function _unpauseMarket(IMToken mtoken) private {
+        vm.startPrank(addresses.getAddress("MOONBEAM_TIMELOCK"));
+        comptroller._setBorrowPaused(mtoken, false);
+        vm.stopPrank();
+    }
+
+    function _fixUser(address _user) private {
+        uint256 cash = delegator.getCash();
+
+        _accrueInterest();
+
+        assertEq(delegator.badDebt(), 0);
+        assertEq(delegator.getAccountTokens(addresses.getAddress("NOMAD_REALLOCATION_MULTISIG")), 0);
+
+        delegator.fixUser(addresses.getAddress("NOMAD_REALLOCATION_MULTISIG"), _user);
+
+        assertEq(delegator.getCash(), (cash + delegator.badDebt()));
+    }
+
+    function testFixUserWithBadDebt() public {
+        uint256 cash = delegator.getCash();
+        uint256 totalBorrows = delegator.totalBorrows();
+        uint256 totalReserves = delegator.totalReserves();
+
+        assertEq(delegator.getAccountTokens(user), 2264582);
+        vm.startPrank(addresses.getAddress("MOONBEAM_TIMELOCK"));
+        _fixUser(user);
+        vm.stopPrank();
+
+        assertEq(delegator.getCash(), (cash + delegator.badDebt()));
+        assertTrue(delegator.totalBorrows() < totalBorrows);
+        assertTrue(delegator.totalReserves() > totalReserves);
+        assertEq(delegator.totalSupply(), 13286905495267783);
+        assertEq(delegator.badDebt(), 357392405781480063721876);
+
+        assertEq(delegator.getAccountTokens(addresses.getAddress("NOMAD_REALLOCATION_MULTISIG")), 2264582);
```

there should not be any hardcoded values here, otherwise once the proposal passes, things will fail all over the place

---

I guess in this case, it's fine because these values aren't changing, but maybe move the constants into a variable at the top of the file to keep track of things.

---

Updated as per our earlier discussion.